posix/signal.c

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/signal.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/unistd.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/print.h>
+#include <openenclave/internal/thread.h>
+#include <openenclave/internal/trace.h>
+#include "posix_t.h"
+
+// Poll uses much of the infrastructure from epoll.
+
+static void _handle_ignore(int signum);
+
+static struct oe_sigaction _actions[__OE_NSIG] = {{{0}}};
+static oe_sighandler_t _default_actions[__OE_NSIG] = {
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore};
+
+static void _handle_ignore(int signum)
+{
+    (void)signum;
+}
+
+static void _handle_error(int signum)
+{
+    (void)signum;
+}
+
+int oe_kill(oe_pid_t pid, int signum)
+{
+    int retval = -1;
+    oe_errno = 0;
+
+    if (oe_posix_kill_ocall(&retval, (int)pid, signum) != OE_OK)
+        OE_RAISE_ERRNO(OE_EINVAL);
+
+    retval = 0;
+
+done:
+    return retval;
+}
+
+int oe_sigaction(
+    int signum,
+    const struct oe_sigaction* act,
+    struct oe_sigaction* oldact)
+{
+    int retval = -1;
+
+    oe_register_posix_ecall_function_table();
+
+    if (signum >= __OE_NSIG)
+        OE_RAISE_ERRNO(OE_EINVAL);
+
+    if (oldact)
+    {
+        *oldact = _actions[signum];
+    }
+
```

Comment:
I think this and subsequent functions that modify _actions might need a lock

Reply:
Thanks. Added.